Documentation/Storage-Configuration/Ceph-CSI/ceph-csi-volumegroup-snapshot.md

```diff
@@ -0,0 +1,107 @@
+```
+
+### CephFS VolumeGroupSnapshot
+
+In [VolumeGroupSnapshot](https://github.com/rook/rook/tree/master/deploy/examples/csi/cephfs/groupsnapshot.yaml),
+`volumeGroupSnapshotClassName` should be the name of the `VolumeGroupSnapshotClass`
+previously created. The labels inside `matchLabels` should be present on the
+PVCs that are already created by the CephFS CSI driver.
+
+```console
+kubectl create -f deploy/examples/csi/cephfs/groupsnapshot.yaml
+```
+
+### Verify CephFS GroupSnapshot Creation
+
+```console
+$ kubectl get volumegroupsnapshotclass
+NAME                              DRIVER                          DELETIONPOLICY   AGE
+csi-cephfsplugin-groupsnapclass   rook-ceph.cephfs.csi.ceph.com   Delete           21m
+```
+
+```console
+$ kubectl get volumegroupsnapshot
+NAME                       READYTOUSE   VOLUMEGROUPSNAPSHOTCLASS          VOLUMEGROUPSNAPSHOTCONTENT                              CREATIONTIME   AGE
+cephfs-pvc-groupsnapshot   true         csi-cephfsplugin-groupsnapclass   groupsnapcontent-d13f4d95-8822-4729-9586-4f222a3f788e   5m37s          5m39s
+```
+
+The snapshot will be ready to restore to a new PVC when `READYTOUSE` field of the
+`volumegroupsnapshot` is set to true.
+
+### Restore the CephFS groupsnapshot to a new PVC
+
+We need to find the name of the snapshots created by the `VolumeGroupSnapshot` first by running:
+
+```console
+$ kubectl get volumegroupsnapshot/cephfs-pvc-groupsnapshot -o=jsonpath='{range .status.pvcVolumeSnapshotRefList[*]}PVC: {.persistentVolumeClaimRef.name}, Snapshot: {.volumeSnapshotRef.name}{"\n"}{end}'
+PVC: cephfs-pvc, Snapshot: snapshot-9d21b143904c10f49ddc92664a7e8fe93c23387d0a88549c14337484ebaf1011-2024-09-12-3.49.13
+```
+
+It will list the PVC's name followed by its snapshot name.
+
+In
+[pvc-restore](https://github.com/rook/rook/tree/master/deploy/examples/csi/cephfs/pvc-restore.yaml),
+`dataSource` should be one of the `Snapshot` that we just
+found. The `dataSource` kind should be the `VolumeSnapshot`.
+
+Create a new PVC from the snapshot
+
+```console
+kubectl create -f deploy/examples/csi/cephfs/pvc-restore.yaml
+```
+
+### Verify CephFS Restore PVC Creation
+
+```console
+$ kubectl get pvc
+cephfs-pvc           Bound    pvc-9ae60bf9-4931-4f9a-9de1-7f45f31fe4da   1Gi        RWO            rook-cephfs    <unset>                 171m
+cephfs-pvc-restore   Bound    pvc-b4b73cbb-5061-48c7-9ac8-e1202508cf97   1Gi        RWO            rook-cephfs    <unset>                 46s
+```
+
+## CephFS volume group snapshot resource Cleanup
+
+To clean your cluster of the resources created by this example, run the following:
+
+```console
+kubectl delete -f deploy/examples/csi/cephfs/pvc-restore.yaml
+kubectl delete -f deploy/examples/csi/cephfs/groupsnapshot.yaml
+kubectl delete -f deploy/examples/csi/cephfs/groupsnapshotclass.yaml
+```
+
+## Limitations
+
+- The PVC cannot be deleted if it has snapshots. make sure all the snapshots on the PVC are deleted before you delete the PVC.
```

Comment:
```suggestion
- The PVC cannot be deleted if it has snapshots. All snapshots on the PVC must be deleted before the PVC is deleted.
```

Reply:
we should remove this limitation from both group snapshot and snapshot, we have the flexibility to remove snapshot after the clone is done